src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala

```diff
@@ -71,17 +72,22 @@ class ConsignmentService(
       sequence <- consignmentRepository.getNextConsignmentSequence
       body <- transferringBodyService.getBodyByCode(userBody)
       consignmentRef = ConsignmentReference.createConsignmentReference(yearNow, sequence)
+      consignmentId = uuidSource.uuid
       consignmentRow = ConsignmentRow(
-        uuidSource.uuid,
+        consignmentId,
         addConsignmentInput.seriesid,
         token.userId,
-        Timestamp.from(now),
+        timestampNow,
         consignmentsequence = sequence,
         consignmentreference = consignmentRef,
         consignmenttype = consignmentType,
         bodyid = body.bodyId
       )
+      descriptiveMetadataStatusRow = ConsignmentstatusRow(uuidSource.uuid, consignmentId, DescriptiveMetadata, "NotEntered", timestampNow, Option(timestampNow))
+      closureMetadataStatusRow = ConsignmentstatusRow(uuidSource.uuid, consignmentId, ClosureMetadata, "NotEntered", timestampNow, Option(timestampNow))
       consignment <- consignmentRepository.addConsignment(consignmentRow).map(row => convertRowToConsignment(row))
+      _ <- consignmentStatusRepository.addConsignmentStatus(descriptiveMetadataStatusRow)
+      _ <- consignmentStatusRepository.addConsignmentStatus(closureMetadataStatusRow)
```

Comment:
We may create another method `addConsignmentStatuses` which will accept statuses as a list and pass `descriptiveMetadataStatusRow` and `closureMetadataStatusRow` as a list to do the transition in a single call.

Reply:
Good idea, I've added the method in and a repository test for it.